pages/instructorAssessmentGroups/instructorAssessmentGroups.js

```diff
@@ -74,9 +74,21 @@ router.post('/', function(req, res, next) {
             res.redirect(res.locals.urlPrefix + '/jobSequence/' + job_sequence_id);
         });
     } else if (req.body.__action == 'copy_assessment_groups') {
+        //make sure the copying assessment_ belong to the same course_instance
+        let sameInstance = false;
+        res.locals.assessments.forEach(element => {
+            if (element.id == req.body.copyAssessmentId) {
+                sameInstance = true;
+            }
+        });
+        if (!sameInstance) {
+            res.locals.errormsg = 'The assessment that you are copying does not belong to the same course instance. Please select another one.';
+            obtainInfo(req, res, next);
+            return;
+        }
```

Comment:
I would suggest deleting all these checks in the JS code and instead add a check in the `assessment_groups_copy` sproc. When you look up the `temp_copy_group_config_id` you can do something like:
```
SELECT gc.id INTO temp_copy_group_config_id
FROM
    assessments AS src_a
    JOIN group_configs AS src_gc ON (src_gc.assessment_id = src_a.id)
    JOIN assessments AS dest_a ON (dest_a.id = assessment_id)
WHERE
    src_a.id = copying_assessment_id
    AND src_a.course_instance_id = dest_a.course_instance_id   -- check they are in the same course instance
    AND src_a.deleted_at IS NULL
    AND gc.deleted_at IS NULL;
```

After this you can do:
```
IF NOT FOUND THEN
    RAISE EXCEPTION 'Could not find source assessment';
END IF;
```